### PR TITLE
[feat]: [CDS-39942]: added yaml in environement request

### PR DESCRIPTION
**Pull request by tchaurasiya**

### CODE REVIEW

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-go-sdk/170)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @micahlmartin and @rathodmeetsatish)

File: harness/nextgen/model_environment_request.go [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
+	// Tags
+	Tags map[string]string `json:"tags,omitempty"`
+	// Name of the Environment.
+	Name string `json:"name,omitempty"`
+	// Description of the entity
+	Description string `json:"description,omitempty"`
+	// Color of the Environment.
+	Color string `json:"color,omitempty"`
+	// Specify the environment type whether production or Preproduction.
+	Type_ string `json:"type,omitempty"`
```
Comment (RajBaviskar) [inline harness/nextgen/model_environment_request.go:29]: 'Type_' this is generated by codegen?

Comment (tchaurasiya) [reply harness/nextgen/model_environment_request.go:29]: Yes this is coming from codegen. But i see the point. This is breaking the implementation that we have made in 
**terraform-provider-harness** repo for **resource_environment.go**

Comment (tchaurasiya) [reply harness/nextgen/model_environment_request.go:29]: I have made the adaptive changes to **terraform-provider-harnes**s repo to support this.

**Pull request merged**